src/server/pegasus_server_impl.cpp

```diff
@@ -1237,6 +1242,90 @@ void pegasus_server_impl::on_scan(const ::dsn::apps::scan_request &request,
 
 void pegasus_server_impl::on_clear_scanner(const int64_t &args) { _context_cache.fetch(args); }
 
+void pegasus_server_impl::on_detect_hotkey(
+    const ::dsn::apps::hotkey_detect_request &args,
+    ::dsn::rpc_replier<::dsn::apps::hotkey_detect_response> &reply)
+{
+    if (args.operation == dsn::apps::hotkey_collector_operation::START) {
+        on_start_detect_hotkey(args, reply);
+    } else if (args.operation == dsn::apps::hotkey_collector_operation::STOP) {
+        on_stop_detect_hotkey(args, reply);
+    } else {
+        derror("unkwnon hotkey_collector_operation");
+        ::dsn::apps::hotkey_detect_response resp;
+        resp.err = ::dsn::ERR_SERVICE_NOT_FOUND;
+        reply(resp);
+    }
+}
+
+void pegasus_server_impl::on_start_detect_hotkey(
+    const ::dsn::apps::hotkey_detect_request &args,
+    ::dsn::rpc_replier<::dsn::apps::hotkey_detect_response> &reply)
+{
+    bool is_read(false);
+    ::dsn::apps::hotkey_detect_response resp;
+    std::shared_ptr<hotkey_collector> hotkey_collector_pointer;
+
+    if (args.type == dsn::apps::hotkey_type::READ) {
+        is_read = true;
+    } else if (args.type == dsn::apps::hotkey_type::WRITE) {
+        is_read = false;
+    } else {
+        derror("Error hotkey_type");
+        resp.err = ::dsn::ERR_SERVICE_NOT_FOUND;
+        reply(resp);
+    }
+
+    if (is_read) {
```

Comment:
Is it allowed _read_hotkey_collector and _write_hotkey_collector to work concurrently?